src/main/java/course_project/Main.java

```diff
@@ -0,0 +1,9 @@
+package course_project;
+
+public class Main {
+
+    public static void main(String[] args) {
+        new Game().run();
```

Comment:
Good! The game works fine, even though it's not really comfortable to play. I didn't find any bugs, and you show good knowledge and skills using Java Core and Java libraries, Java 8 features as well. 

To make better: refactor interface (automatic ship placement, field representation to more comfortable. You chose the easiest approach where fields representation is the same for both players, it could be improved.)
Also, here and there I can see typos and Code Conventions violations in minor things. Not critical, but disturbing.